utils/parsedate.py

```diff
@@ -0,0 +1,30 @@
+import sys
+import re
+from datetime import datetime
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2:
+        raise TypeError("Must pass file to parse as an argument")
+
+    file_to_parse = sys.argv[1]
+    max_timestamp = None
+
+    with open(file_to_parse, "r") as f:
+        for line in f:
+            # split line with whitespace, '=', and '>' deliminators
+            split_line = re.split("[\s=>]", line)
+            # timestamp will be at index next to 'timestamp' string
+            if "timestamp" in split_line:
+                timestamp_idx = split_line.index("timestamp") + 1
+                timestamp_str = split_line[timestamp_idx]
```

Comment:
This will work with all current XML diff files, so probably fine.

More robust (and much shorter) is a simple regex-search: `line.search(r' timestamp="([^"]+)"').group(1)`